_posts/2018-05-25-support-process.markdown

```diff
@@ -0,0 +1,99 @@
+
+Our ad-hoc support practice was working, but also had some clear drawbacks:
+
+1.  Answering support issues took a lot of time/focus away from engineers working on sprint tasks, and this responsibility was not distributed
+2.  Bugs and maintenance items weren’t always prioritized or worked on consistently.
+3.  Knowledge about how systems work or how decisions were made was siloed in individuals or small teams
+4.  Issues related to features that were not “owned” by an obvious product area often went ignored
+
+We wanted to create a system that would ensure stable, constantly improving services in spite of all of Artsy’s growing and changing.
+
+# Goals/Research
+
+We first identified the goals of our ideal support process:
+
+1.  People who report issues have full confidence that their issue will be solved, or understand why it won’t be.
+2.  Engineers feel empowered and able to fix issues and understand when it is their responsibility to do so.
+3.  We evolve our systems to require less support and spread knowledge among engineers so there are fewer bottlenecks.
+4.  All consumers of Artsy (employees, users, and partners) feel confident that our systems are stable and we will address any issues that arise.
+
+With those goals in mind, we researched both externally (looking at how other companies tackled this common problem) and internally (talking to members of the engineering team and various business/product stakeholders).
+
+Out of this effort, we learned that people had some shared concerns:
+
+* The same few people tended to address most of the support questions, meaning the responsibility was imbalanced.
+* People felt wary about the idea of having to answer to support requests from areas they were unfamiliar with.
+* People wondered how a support duty might fit into their product work.
+* Without a dedicated process people worried that even if bugs were identified they might not be fixed (thus contributing to future support needs).
+
+We then identified a few potential solutions:
+
+* We could hire or dedicate people to handle support as a full-time job.
+* We could have a rotating set of engineers who are exempt from product-related work and tasked with both answering immediate support requests and fixing small bugs or improving the state of our infrastructure/monitoring.
+* We could have engineers on an on-call rotation who are responsible for addressing immediate issues but remain engaged with their product team.
+
+# Our Plan
+
+We worked with the product management team to identify an escalation path for different types of issues. At the least-urgent level were basic feature requests, and at the most-urgent were critical bugs or “incidents”. We decided to focus on building out a process whereby engineers could be staffed to always address urgent incidents, and other, less-urgent bugs or issues could be triaged and added to teams’ workloads as part of a normal sprint.
```

Comment:
_...whereby certain engineers could be available to address urgent incidents..._